src/AutorestSwift/View Models/ResponseViewModel.swift

```diff
@@ -61,14 +61,21 @@ enum ResponseBodyType: String {
             case .dateTime:
                 if let dateTimeSchema = schema as? DateTimeSchema,
                     dateTimeSchema.format == .dateTimeRfc1123 {
-                    return .datetimeRfc1123Body
+                    return .dateTimeRfc1123Body
                 }
                 return .dateTimeBody
             default:
                 return .dateBody
             }
-        } else if input == "Data", type == .byteArray {
+        } else if type == .byteArray {
             return .byteArrayBody
+        } else if input == "[Date]" {
+            if let arraySchema = schema as? ArraySchema,
+                let dateTimeSchema = arraySchema.elementType as? DateTimeSchema {
+                return (dateTimeSchema.format == .dateTimeRfc1123) ? .dateTimeRfc1123Body : .dateTimeBody
+            } else {
+                return .dateBody
+            }
```

Comment:
Can an array of dates not have a custom format?

Reply:
As far as i know, just 3 formats: Date, DateTime (iso8601), DateTime (rfc 1123)

Reply:
Certain good for now.